src/nti/contentfragments/html.py

```diff
@@ -194,8 +194,12 @@ def __iter__(self):
                     yield token
 
     def _find_links_in_text(self, token):
+        if self.link_finder is None:
+            yield token
+            return
+            
         text = token['data']
-        text_and_links = self.link_finder.find_links(text) if self.link_finder else ()
+        text_and_links = self.link_finder.find_links(text)
```

Comment:
Instead of introducing a new shortcut code path, given that there's already a conditional immediately following this, that, when it fails, just yields the token (exactly what the new condition is doing), I would probably have changed the constructor: `self.link_finder = component.queryUtility(IHyperlinkFormatter, dummy_formatter)`, where  `dummy_formatter` looks like:
```python
class _DummyFormatter(object):
    def find_links(self, text):
       # Need to return a single element; this is interpreted as not actually finding links
        return (text,)
dummy_formatter = _DummyFormatter()
```
(Choosing better names, of course.)

That also removes the condition from the original line (`if self.link_finder else ()`) because we can always assume `link_finder` is non-None.

An even simpler change would be `if self.link_finder else (text,)`, but that still leaves the condition.